pkg/nodeagent/controller/operatingsystemconfig/containerd.go

```diff
@@ -0,0 +1,170 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package operatingsystemconfig
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pelletier/go-toml"
+	"k8s.io/utils/pointer"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/retry"
+)
+
+// ReconcileContainerdConfig sets required values of the given containerd configuration.
+func (r *Reconciler) ReconcileContainerdConfig(ctx context.Context, log logr.Logger, containerdChanges containerd) error {
+	if err := r.ensureContainerdConfigDirectories(); err != nil {
+		return err
```

Comment:
Nit: It is also a good idea to wrap the error for debugging purposes.

```suggestion
		return fmt.Errorf("failed to ensure containerd config directories: %w", err)
```